stories/text/text.stories.js

```diff
@@ -0,0 +1,159 @@
+    quisque id diam vel quam elementum pulvinar. Eget aliquet nibh praesent tristique magna. Id interdum velit laoreet
+    id donec ultrices tincidunt arcu. Suspendisse ultrices gravida dictum fusce ut placerat orci nulla. Montes nascetur
+    ridiculus mus mauris vitae ultricies. In nisl nisi scelerisque eu. Suspendisse interdum consectetur libero id
+    faucibus. Ut lectus arcu bibendum at varius. Placerat vestibulum lectus mauris ultrices eros in cursus turpis massa.
+    Dolor magna eget est lorem ipsum dolor sit. Aliquam nulla facilisi cras fermentum odio eu feugiat pretium nibh.
+    Augue neque gravida in fermentum et. Proin fermentum leo vel orci porta non pulvinar.</p>
+`;
+
+maxLines.storyName = 'Max lines';
+maxLines.parameters = {
+    docs: {
+        iframeHeight: 300,
+        storyDescription: `You can define maximum number of lines for text, after reaching the limit text will be
+        truncated and ellipsis will be added. To set maximum numbers you should add
+        <code class="docs-code">fd-text--max-lines</code> class and inline style rule
+        <code class="docs-code">-webkit-line-clamp</code> with number of visible lines. For example:
+        <code class="docs-code">style="-webkit-line-clamp: 3;"</code>`
+    }
+};
+
+export const hyphenation = () => `<div class="example-container">
+        <div class="mw-300">
+            <h3>Without hyphens</h3>
+            <p class="fd-text">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
+            ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+            aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore
+            eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia
+            deserunt mollit anim id est laborum.</p>
+        </div>
+
+        <div class="mw-300">
+            <h3>Hyphenation</h3>
+            <p class="fd-text fd-text--hyphenation">Lorem ipsum dolor sit amet, con&shy;sectetur adip&shy;iscing el&shy;it, sed do eiusmod tempor
+            incididunt ut labore et do&shy;lore magna aliqua. Ut enim ad mi&shy;nim veniam, quis nostrud exer&shy;citation ullamco laboris nisi ut
+            aliquip ex ea commodo conse&shy;quat. Duis aute irure dolor in repre&shy;henderit in vo&shy;luptate velit esse cillum dolore
+            eu fu&shy;giat nulla pari&shy;atur. Excepteur sint occaecat cupidatat non pro&shy;ident, sunt in culpa qui officia
+            deserunt mollit anim id est la&shy;bo&shy;rum.</p>
+        </div>
+    </div>
+`;
+
+hyphenation.storyName = 'Hyphenation';
+hyphenation.parameters = {
+    docs: {
+        iframeHeight: 200,
+        storyDescription: `To allow browser to automatically break words at appropriate hyphenation points, 
```

Comment:
The text component can display words that are broken at appropriate hyphenation points in a text block. To display hyphens, add the <code class="docs-code">fd-text--hyphenation</code> to the main element. 

It is also possible to suggest line break opportunities with two Unicode characters that manually specify potential line breakpoints:
- Hyphen: The "hard" hyphen character indicates a visible line break opportunity. Even if the line is not actually broken at that point, the hyphen is still displayed.
- Shy: An invisible, "soft" hyphen. Although this character is not visible, it marks a place where the browser should break the word if hyphenation is necessary.

In HTML, add the `&shy;` Unicode to insert a soft hyphen.

Reply:
changed